lib/puma/cluster/pipe_protocols.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Puma
+  class Cluster
+    module PipeProtocols
+      module Fork
```

Comment:
this is mostly a nice-to-have, on its own it did not prevent the root cause but it makes it _much_ safer to replace mold processes without complex draining logic